- **Kelas** : IF403
- **Program Studi** : PJJ Informatika
- **Nama Mahasiswa** : Raka Anggie Saputra
- **NIM** : 240401010148

In [ ]:
import pandas as pd
import numpy as np

In [ ]:
# Muat Dataset Titanic
URL = "https://raw.githubusercontent.com/datasciencedojo/datasets/master/titanic.csv"
df = pd.read_csv(URL)

In [ ]:
# Eksplorasi Awal
print(df.shape, df.columns.tolist())
print(df.isnull().sum())

(891, 12) ['PassengerId', 'Survived', 'Pclass', 'Name', 'Sex', 'Age', 'SibSp', 'Parch', 'Ticket', 'Fare', 'Cabin', 'Embarked']
PassengerId      0
Survived         0
Pclass           0
Name             0
Sex              0
Age            177
SibSp            0
Parch            0
Ticket           0
Fare             0
Cabin          687
Embarked         2
dtype: int64


In [ ]:
# Filtering – Penumpang Wanita Kelas 1
wk1 = df[(df['Sex']=='female') &
         (df['Pclass']==1)]
print(wk1['Survived'].mean(), wk1['Age'].mean())

0.9680851063829787 34.61176470588235


In [ ]:
# Groupby – Tingkat Keselamatan per Kelas
hasil = df.groupby('Pclass').agg(
    total=('PassengerId','count'),
    selamat=('Survived','mean'),
).round(3)
print(hasil)

        total  selamat
Pclass                
1         216    0.630
2         184    0.473
3         491    0.242


# Deskripsi Notebook - Pertemuan 2

## Rangkuman Isi Notebook

Notebook ini berfokus pada teknik *Data Wrangling* awal menggunakan repositori data publik. Berikut adalah poin-poin analisis yang dilakukan di dalamnya:

1. **Inisialisasi Pustaka (Library Import)**
   * Mengimpor pustaka utama analisis data, yaitu **Pandas** (`pd`) dan **NumPy** (`np`).

2. **Pemuatan Dataset (Data Loading)**
   * Mengambil dan membaca Dataset Titanic berformat CSV secara langsung dari repositori GitHub eksternal menggunakan fungsi `pd.read_csv()`.

3. **Eksplorasi Awal Data (Initial Exploration)**
   * Memeriksa dimensi data (`df.shape`) yang menunjukkan adanya **891 baris dan 12 kolom**, serta mendaftar seluruh nama kolom yang tersedia.
   * Melakukan pengecekan nilai yang hilang (`df.isnull().sum()`), di mana ditemukan data kosong pada kolom *Age* (177 baris), *Cabin* (687 baris), dan *Embarked* (2 baris).

4. **Penyaringan Data (Filtering)**
   * Melakukan filter spesifik untuk segmentasi **Penumpang Wanita Kelas 1** (`Sex == 'female'` & `Pclass == 1`).
   * Menghitung metrik rata-rata keselamatan (~96.8%) dan rata-rata usia (~34.6 tahun) khusus pada segmen tersebut.

5. **Pengelompokan Data (Groupby & Aggregation)**
   * Mengelompokkan data berdasarkan kelas tiket (*Pclass*) menggunakan fungsi `.groupby()`.
   * Melakukan agregasi statistik untuk menghitung **total penumpang** dan **tingkat keselamatan (mean)** pada masing-masing kelas, lalu membulatkan hasilnya hingga 3 angka di belakang koma.

## Status Eksekusi
Seluruh blok kode di dalam notebook ini telah **berhasil dijalankan dengan sukses (runtime ok)** dan menampilkan keluaran data statistik yang valid secara langsung di bawah sel eksekusi.